tests/Acoustics.Test/AudioAnalysisTools/DSP/UnsupervisedFeatureLearningTest.cs

```diff
@@ -0,0 +1,407 @@
+                allMaxFeatureVectors.Add(maxFeatureVectors.ToArray().ToMatrix());
+                allStdFeatureVectors.Add(stdFeatureVectors.ToArray().ToMatrix());
+            }
+
+            //+++++++++++++++++++++++++++++++++++Temporal Summarization
+
+            //++++++++++++++++++++++++++++++++++Writing features to file
+            //First, concatenate mean, max, std for each second.
+            //Then write to CSV file.
+
+            for (int j = 0; j < allMeanFeatureVectors.Count; j++)
+            {
+                var outputFeatureFile = Path.Combine(resultDir, "FeatureVectors" + j.ToString() + ".csv");
+
+                //creating the header for CSV file
+                List<string> header = new List<string>();
+                for (int i = 0; i < allMeanFeatureVectors.ToArray()[j].GetLength(1); i++)
+                {
+                    header.Add("mean" + i.ToString());
+                }
+
+                for (int i = 0; i < allStdFeatureVectors.ToArray()[j].GetLength(1); i++)
+                {
+                    header.Add("std" + i.ToString());
+                }
+
+                for (int i = 0; i < allMaxFeatureVectors.ToArray()[j].GetLength(1); i++)
+                {
+                    header.Add("max" + i.ToString());
+                }
+
+                //concatenating mean, std, and max vector together for each 1 second
+                List<double[]> featureVectors = new List<double[]>();
+                for (int i = 0; i < allMeanFeatureVectors.ToArray()[j].ToJagged().GetLength(0); i++)
+                {
+                    List<double[]> featureList = new List<double[]>();
+                    featureList.Add(allMeanFeatureVectors.ToArray()[j].ToJagged()[i]);
+                    featureList.Add(allMaxFeatureVectors.ToArray()[j].ToJagged()[i]);
+                    featureList.Add(allStdFeatureVectors.ToArray()[j].ToJagged()[i]);
+                    double[] featureVector = DataTools.ConcatenateVectors(featureList);
+                    featureVectors.Add(featureVector);
+                }
+
+                //writing feature vectors to CSV file
+                using (StreamWriter file = new StreamWriter(outputFeatureFile))
+                {
+                    //writing the header to CSV file
+                    foreach (var entry in header.ToArray())
+                    {
+                        file.Write(entry + ",");
+                    }
+
+                    file.Write(Environment.NewLine);
+
+                    foreach (var entry in featureVectors.ToArray())
+                    {
+                        foreach (var value in entry)
+                        {
+                            file.Write(value + ",");
+                        }
+
+                        file.Write(Environment.NewLine);
+                    }
+                }
+            }
+
+            List<double[,]> convertedSpec = new List<double[,]>();
```

Comment:
This unit test has no assertions (and it is very long). Perhaps it needs to be redesigned?

Reply:
Actually, this is not a unit test. This will be a part of IAnalyser. Currently, I am using this as entry point for the experiments I need to run.